### PR TITLE
Mtdsa 615

### PR DESCRIPTION
**Pull request by hicnar**

### CODE REVIEW

File: test/uk/gov/hmrc/selfassessmentapi/repositories/domain/calculations/LiabilitySpec.scala [hunk 1]
```diff
@@ -506,12 +506,12 @@ class LiabilitySpec extends UnitSpec {
 
 case class ComputeLiabilityFor(employments: Seq[Employment] = Nil, selfEmployments: Seq[SelfEmployment] = Nil,
                                ukProperties: Seq[UKProperties] = Nil, unearnedIncomes: Seq[UnearnedIncome] = Nil,
-                               furnishedHolidayLettings: Seq[FurnishedHolidayLettings] = Nil, banks: Seq[Bank] = Nil) {
+                               furnishedHolidayLettings: Seq[FurnishedHolidayLettings] = Nil,
+                               dividends: Seq[MongoDividend] = Nil,  banks: Seq[Bank] = Nil) {
```
Comment (GlassAndOneHalf) [inline test/uk/gov/hmrc/selfassessmentapi/repositories/domain/calculations/LiabilitySpec.scala]: Why `MongoDividend` and not `Dividend`? We should be hiding the fact we use mongo this far up.

File: test/uk/gov/hmrc/selfassessmentapi/repositories/live/DividendsRepositorySpec.scala [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  private val summariesMap: Map[JsonMarshaller[_], SummaryRepository[_]] = Map(DividendIncome -> mongoRepository.DividendIncomeRepository)
+
+  override def beforeEach() {
+    await(mongoRepository.drop)
+    await(mongoRepository.ensureIndexes)
+  }
+
+  val saUtr = generateSaUtr()
+
+  def unearnedIncome(): Dividend = Dividend.example()
```
Comment (GlassAndOneHalf) [inline test/uk/gov/hmrc/selfassessmentapi/repositories/live/DividendsRepositorySpec.scala]: You can remove a lot of the type annotations in this file, they just add noise imo.

File: test/uk/gov/hmrc/selfassessmentapi/repositories/live/DividendsRepositorySpec.scala [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    "add a summary to an empty list when source exists and return id" in {
+      for ((summaryItem, repo) <- summariesMap) {
+        val sourceId = await(unearnedIncomeMongoRepository.create(saUtr, taxYear, unearnedIncome()))
+        val summary = summaryItem.example()
+        val summaryId = await(repo.create(saUtr, taxYear, sourceId, cast(summary)))
+
+        summaryId.isDefined shouldEqual true
+        val dbSummaries = await(repo.list(saUtr, taxYear, sourceId))
+
+        val found = dbSummaries.get
```
Comment (GlassAndOneHalf) [inline test/uk/gov/hmrc/selfassessmentapi/repositories/live/DividendsRepositorySpec.scala]: Inline this get on line 147.

**Comment by rakeshavasarala** [comment]
OK to merge from testing.

**Pull request merged**